### ISSUE TITLE
cwltool enumerates every single file and folder in the output directory, unneccesarily

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/cwltool#561]
https://github.com/common-workflow-language/cwltool/blob/master/cwltool/process.py#L310

To reproduce: run `cwltool --outdir ${HOME}` and a simple CWL description on real user's system

Perhaps a single temporary folder should be created within `outdir` where the intermediate output directories are then stored under? That way this loop would iterate over just that single folder instead of walking the entire tree from `outdir`.

**Comment by mr-c** [comment]
@esanzgar you are correct, I didn't use a permalink so the line numbers shifted. I've updated the issue, thanks!

**Comment by psafont** [comment]
Work happening on https://github.com/common-workflow-language/cwltool/pull/850

**Comment by psafont** [comment]
Solved with https://github.com/common-workflow-language/cwltool/pull/926
